src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs

```diff
@@ -66,11 +87,9 @@ public Task<RestoreResult> ExecuteAsync()
 
         public async Task<RestoreResult> ExecuteAsync(CancellationToken token)
         {
-            using (var telemetry = TelemetryActivity.CreateTelemetryActivityWithNewOperationId(ParentId))
+            using (var telemetry = TelemetryActivity.CreateTelemetryActivityWithNewOperationIdAndEvent(parentId: ParentId, eventName: ProjectRestoreInformation))
```

Comment:
Is it possible to modularize this code instead of having everything in that single API? I feel it has made this `ExecuteAsync` more complex and unreadable.